### ISSUE TITLE
Product Collection: Hide preview label when "From a specific product" is selected

### ISSUE THREAD

**Issue by imanish003** [woothemes/woocommerce#51601]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Currently, in the Product Collection block, the preview label is visible, even when "From a specific product" is selected. When a specific product is chosen, the preview label should be hidden as we know the exact product reference.

### Expected behavior

When "From a specific product" is selected and a product is chosen, the preview label should be hidden as we know the exact product reference.

### Actual behavior

The preview label remains visible even when a specific product is selected.

### Steps to reproduce

1. Add a Product Collection block in Single Product template
2. Choose "Upsells" collection
3. In the block settings, change the product reference option from "From the current product" to "From a specific product".
4. Select a specific product.
5. Observe that the preview label is still visible.

### WordPress Environment

-

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by octaedro** [comment]
Hi @imanish003,

I'm taking a look at this issue an I have a few questions.

Maybe I didn't understand the expected behavior for that selector. Under the `Products to show` options, there are two options: `From a specific product` and `From the current product`. However, I couldn't find any differences after selecting one or another.

✅ - After selecting `From a specific product` I see the Linked product of the product I'm seeing, that's correct.
❌ - But when I select `From the current product` and choose another product, I continue seeing the same products, I would expect to see the linked product of the product I selected. 

https://github.com/user-attachments/assets/2d8328f8-f622-4435-8998-76083acf82b8

If the expected behavior is to change the items shown in the `You may also like` section after selecting another product's linked products, I think we should address that before working on this issue.

**Comment by imanish003** [comment]
Good catch! You're right, the product should change based on selected product. Which doesn't seem to be the case. I think we should address it before WooCommerce 9.5 Feature Freeze (October 21, 2024).
1. When "From the current product" is selected, it should display the linked products of the current product being viewed.
2. When "From a specific product" is selected and a product is chosen, it should display the linked products of that specific product.

CC: @kmanijak @ObliviousHarmony Looks like we weren't able to catch it as we merged [this PR](https://github.com/woocommerce/woocommerce/pull/51359) in parallel to new collections PRs.

**Comment by imanish003** [comment]
Update: I was able to debug the root cause of this issue. It looks like there is a typo in a variable name. I'm going to create a PR to fix it.

**Comment by imanish003** [comment]
Update: I [fixed](https://github.com/woocommerce/woocommerce/pull/51730) the Upsells issue for frontend as it's important that shopper facing pages are fixed. We can create a follow-up PR to fix the issue on Editor. 

**Next Steps:** Both Upsells and Related Products collection should show products based on selected product on Editor side in Preview mode. For now, we aren't handling preview mode for both of these collections.
